clients/client-python/tests/integration/test_gvfs_with_gcs.py

```diff
@@ -0,0 +1,173 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import logging
+import os
+from random import randint
+import unittest
+
+from fsspec.implementations.arrow import ArrowFSWrapper
+from pyarrow.fs import GcsFileSystem
+
+from tests.integration.test_gvfs_with_hdfs import TestGvfsWithHDFS
+from gravitino import (
+    gvfs,
+    GravitinoClient,
+    Catalog,
+    Fileset,
+)
+from gravitino.exceptions.base import GravitinoRuntimeException
+
+
+logger = logging.getLogger(__name__)
+
+
+@unittest.skip("This test require GCS service account key file")
+class TestGvfsWithGCS(TestGvfsWithHDFS):
+    # Before running this test, please set the make sure gcp-bundle-x.jar has been
+    # copy to the $GRAVITINO_HOME/catalogs/hadoop/libs/ directory
+    key_file = "your_key_file.json"
+    bucket_name = "your_bucket_name"
+    metalake_name: str = "TestGvfsWithGCS_metalake" + str(randint(1, 10000))
+
+    def setUp(self):
+        self.options = {"gravitino.bypass.gcs.service-account-key-path": self.key_file}
+
+    def tearDown(self):
+        self.options = {}
+
+    @classmethod
+    def setUpClass(cls):
+        cls._get_gravitino_home()
+
+        cls.hadoop_conf_path = f"{cls.gravitino_home}/catalogs/hadoop/conf/hadoop.conf"
```

Comment:
If we do not need to append some hadoop configurations in the server side, is there having any necessity to reset hadoop conf here?

Reply:
> If we do not need to append some hadoop configurations in the server side, is there having any necessity to reset hadoop conf here?

I'm afraid things may not be so ideal, I've encountered several cases where there are already some configuration keys in the conf file added by HDFS test cases, for example, we terminated the tests halfway and the file content can't be properly reset, so I will always rewrite it to ensure that the configuration file is clean.

Reply:
I see.